internal/uploader/auth_utils.go

```diff
@@ -1,73 +1,128 @@
+			fmt.Errorf("failed to get workload creds: %w", workloadErr),
+		)
 	}
-	pullSecret, err := validations.ParsePullSecret(data)
+
+	return ret, nil
+}
+
+// Checks if cloud.openshift.com is present in the OCM pull secret to see if the user
+// is still opted-in to sending data
+func isOCMPullSecretOptIn(k8sclient k8sclient.K8SClient) (bool, error) {
+	creds, err := getManagementCreds(k8sclient)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to parse pull secret")
+		if apierrors.IsNotFound(err) {
+			// Indicates the pull secret doesn't exist, so we'll assume they're opted-in
+			return true, nil
+		}
+
+		return false, fmt.Errorf("failed to get management creds: %w", err)
 	}
-	if auth, ok := pullSecret[key]; ok {
-		return &auth, nil
+
+	if creds == nil {
+		return false, nil
 	}
-	return nil, errors.Errorf("pull secret doesn't contain authentication information for %s", key)
+
+	return true, nil
 }
 
-func getOCMPullSecret(k8sclient k8sclient.K8SClient) (string, bool) {
-	if k8sclient != nil {
-		secret, err := k8sclient.GetSecret("openshift-config", "pull-secret")
-		if err != nil {
-			err = client.IgnoreNotFound(err)
-			// A "not found" error (err == nil) should return false
-			// indicating that the secret does not exist
-			return "", err != nil
-		}
-		if pullSecret, ok := secret.Data[corev1.DockerConfigJsonKey]; ok {
-			return string(pullSecret), true
-		}
-		return "", true
+func getManagementCreds(k8sclient k8sclient.K8SClient) (*validations.PullSecretCreds, error) {
+	if k8sclient == nil {
+		return nil, errors.New("nil kube client") // Should not happen
+	}
+
+	secret, err := k8sclient.GetSecret(openshiftNamespace, openshiftSecretName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secret: %w", err)
+	}
+
+	pullSecret, ok := secret.Data[corev1.DockerConfigJsonKey]
+	if !ok {
+		return nil, errors.New("missing .dockerconfigjson field")
+	}
+
+	ret, err := getCloudOpenshiftCreds(string(pullSecret))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cloud openshift creds: %w", err)
+	}
+
+	return ret, nil
+}
+
+func getCloudOpenshiftCreds(pullSecretData string) (*validations.PullSecretCreds, error) {
+	pullSecret, err := validations.ParsePullSecret(pullSecretData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse pull secret: %w", err)
+	}
+
+	auth, ok := pullSecret[openshiftTokenKey]
+	if !ok {
+		return nil, fmt.Errorf("pull secret doesn't contain authentication information for %s", openshiftTokenKey)
+	}
+
+	return &auth, nil
+}
+
+func selectPullSecretForDataProcessing(pullSecrets PullSecretCreds) *validations.PullSecretCreds {
+	if pullSecrets.WorkloadCreds != nil {
+		return pullSecrets.WorkloadCreds
 	}
-	return "", false
+
+	return pullSecrets.ManagementCreds
+}
+
+func selectPullSecretForAuthent(pullSecrets PullSecretCreds) *validations.PullSecretCreds {
+	if pullSecrets.ManagementCreds != nil {
+		return pullSecrets.ManagementCreds
+	}
+
+	return pullSecrets.WorkloadCreds
```

Comment:
Don't these do the same thing and are used the same way?

Reply:
Those 2 methods gathers the logic of choosing the creds for enrichment vs choosing the creds for authN. They just differ on what is the first value (~default value) to use

I like to isolate this kind of logic in dedicated methods so that we are clear about the expected behavior